### PR TITLE
Don't init the store and websocket on hot reload

### PR DESCRIPTION
**Pull request by jcmorrow**
This is.... not the most defensive code I have ever written. Should we use a more obscure key, or is that some other reliable way of knowing that this has already been called?

### CODE REVIEW

File: src/ui/setup/dynamic/devtools.ts [hunk 1]
```diff
@@ -106,39 +107,42 @@ export default async (store: Store) => {
...
-  setupTimeline(store);
-  setupEventListeners(store);
-  setupGraphics(store);
-  initOutputSyntaxHighlighting();
-  setupMessages(store);
-  setupLogpoints(store);
-  setupExceptions(store);
-  setupReactDevTools(store);
+  if (!window.loaded) {
+    window.loaded = true;
```
Comment (jasonLaster) [inline src/ui/setup/dynamic/devtools.ts:111]: I think I'd do two things here:

1. extract to a function
2. more specific variable

```js
function bootstrapApp() {
  if (window.hasAlreadyBootstrapped) {
    return
  } 
    extendStore(store, initialState, reducers, thunkArgs);

    dbgClient.bootstrap(store);

    // Initialize the socket so we can communicate with the server
    initSocket(store, dispatchUrl);
}
```

Comment (jasonLaster) [reply src/ui/setup/dynamic/devtools.ts:111]: also we might need to include this as well

```js
  const settings = await getUserSettings();
  updateEnableRepaint(settings.enableRepaint);
  setupDemo();
```

**Review by jasonLaster — APPROVED** [review]

File: src/ui/setup/dynamic/devtools.ts (hunk 1 above)
Comment (jcmorrow) [reply src/ui/setup/dynamic/devtools.ts:111]: Yeah, actually, this whole function is supposed to be idempotent.

Comment (jcmorrow) [reply src/ui/setup/dynamic/devtools.ts:111]: I think

**Review by jasonLaster — APPROVED** [review]
love it

**Pull request merged**